pkg/kudoctl/cmd/install/params.go

```diff
@@ -4,13 +4,54 @@ import (
 	"errors"
 	"fmt"
 	"strings"
+
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v2"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/util/convert"
 )
 
-// GetParameterMap takes a slice of parameter strings, parses parameters into a map of keys and values
-func GetParameterMap(raw []string) (map[string]string, error) {
+// GetParameterMap takes a slice of parameter strings and a slice of parameter filenames as well as a filesystem,
+// parses parameters into a map of keys and values
+func GetParameterMap(raw []string, filePaths []string, fs afero.Fs) (map[string]string, error) {
 	var errs []string
 	parameters := make(map[string]string)
 
+	for _, filePath := range filePaths {
+		var err error
+		rawData, err := afero.ReadFile(fs, filePath)
+		if err != nil {
+			errs = append(errs, fmt.Sprintf("error reading from parameter file %s: %v", filePath, err))
+			continue
+		}
+		data := make(map[string]interface{})
+		err = yaml.Unmarshal(rawData, data)
+		if err != nil {
+			errs = append(errs, fmt.Sprintf("error unmarshalling content of parameter file %s: %v", filePath, err))
+			continue
+		}
+		for key, value := range data {
+			var valueType v1beta1.ParameterType
+			switch value.(type) {
+			case map[string]interface{}:
+				valueType = v1beta1.MapValueType
+			case []interface{}:
+				valueType = v1beta1.ArrayValueType
+			case string:
+				valueType = v1beta1.StringValueType
+			default:
+				valueType = v1beta1.StringValueType
```

Comment:
It *will* happen whenever a user provides a value of such type in the parameter value file. However since we do not have parameter type (not even parameter name!) information at this stage, we cannot do much about this other than marshal to string, which should DTRT in most cases anyway.

Note that this problem pre-dates this PR. The user can feed any kind of garbage parameter to the CLI and we will happily stuff it into the `Instance`.
I raised this [on slack](https://kubernetes.slack.com/archives/CG3HTFCMV/p1583845701167700) and the consensus so far seems to be to do validation in the API server layer. But I think you are well aware of this since you filed https://github.com/kudobuilder/kudo/issues/1411